frontend/src/views/admin/matching/finalize-changes.tsx

```diff
@@ -0,0 +1,139 @@
+ * @returns
+ */
+export function FinalizeChangesButton() {
+    const [showDialog, setShowDialog] = React.useState(false);
+    const [stagedAssignments, setStagedAssignments] = React.useState<
+        Match[] | null
+    >(null);
+    const matchingData = useSelector(matchingDataSelector);
+    const positions = useSelector(positionsSelector);
+    const applicants = useSelector(applicantsSelector);
+
+    const dispatch = useThunkDispatch();
+
+    React.useEffect(() => {
+        setStagedAssignments(
+            matchingData.matches
+                .filter((match) => match.status === "staged-assigned")
+                .sort((a, b) => {
+                    return `${a.positionCode} ${a.utorid}`.toLowerCase() <
+                        `${b.positionCode} ${b.utorid}`.toLowerCase()
+                        ? -1
+                        : 1;
+                }) || null
+        );
+    }, [matchingData]);
+
+    function onClick() {
+        setShowDialog(true);
+    }
+
+    async function makeAssignment(assignment: Partial<Assignment>) {
+        await dispatch(upsertAssignment(assignment));
+    }
+
+    function _onConfirm() {
+        setShowDialog(false);
+
+        if (!stagedAssignments) {
+            return;
+        }
+
+        for (const match of stagedAssignments) {
+            const targetPosition = positions.find(
+                (position) => position.position_code === match.positionCode
+            );
+            const targetApplicant = applicants.find(
+                (applicant) => applicant.utorid === match.utorid
+            );
+
+            if (!targetPosition || !targetApplicant) {
+                return;
+            }
+
+            const newAssignment: Partial<Assignment> = {
+                position: targetPosition,
+                hours: match.hoursAssigned,
+                applicant: targetApplicant,
+            };
+
+            makeAssignment(newAssignment);
+        }
+    }
+
+    return (
+        <>
+            <Button
+                variant="outline-primary"
+                size="sm"
+                className="footer-button finalize"
+                onClick={onClick}
+                disabled={
+                    stagedAssignments && stagedAssignments.length === 0
+                        ? true
+                        : false
+                }
+            >
+                Finalize Changes{" "}
+                {stagedAssignments && stagedAssignments.length > 0
+                    ? ` (${stagedAssignments.length})`
+                    : ""}
+            </Button>
+            <Modal show={showDialog} dialogClassName="finalize-changes-modal">
+                <Modal.Header>
+                    <Modal.Title>Finalize Changes</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    The following assignments will be made:
+                    {stagedAssignments && stagedAssignments.length > 0 ? (
+                        <ul>
```

Comment:
Use a table for this. Make sure that scrolling is handled correctly when there are 100 staged assignments.